src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -426,23 +426,9 @@ internal AdalResultWrapper LoadFromCacheCommon(CacheQueryData cacheQueryData, Re
 
                         requestContext.Logger.Info("An expired or near expiry token was found in the cache");
                     }
-                    else if (!cacheKey.ResourceEquals(cacheQueryData.Resource))
+                    else if (!tokenIsForSameResource)
                     {
-                        requestContext.Logger.InfoPii(
-                            string.Format(CultureInfo.CurrentCulture,
-                                "Multi resource refresh token for resource '{0}' will be used to acquire token for '{1}'",
-                                cacheKey.Resource, cacheQueryData.Resource),
-                            string.Empty);
-                        var newResultEx = new AdalResultWrapper
-                        {
-                            Result = new AdalResult(null, null, DateTimeOffset.MinValue),
-                            RefreshToken = resultEx.RefreshToken,
-                            ResourceInResponse = resultEx.ResourceInResponse
-                        };
-
-                        newResultEx.Result.UpdateTenantAndUserInfo(resultEx.Result.TenantId, resultEx.Result.IdToken,
-                            resultEx.Result.UserInfo);
-                        resultEx = newResultEx;
+                        resultEx = HandleMrrtCase(cacheQueryData, requestContext.Logger, resultEx, cacheKey);
```

Comment:
This is the part of the bug fix - return null in MRRT case when MRRT is not in cache and request is for a different resource than what we have in the cache.